File: C GCC 5.4.1/ABC/ABC101/ABC101_A_v01.c
// [contest]  ABC 101
// [task]     A
// [URL]      https://atcoder.jp/contests/abc101/tasks/abc101_a
// [compiler] C (GCC 5.4.1)
// [status]   https://atcoder.jp/contests/abc101/submissions/5829377 : AC, with compile errors
// [status]   https://atcoder.jp/contests/abc101/submissions/5829547 : AC

#include <stdio.h>

#define len_string_max 4

int calculate_points (char data_string[len_string_max])
{
    // variables for `calculate_points`
    int points;

    // STEP.01
    // initialize the points
    points = 0;

    // STEP.02
    // calculate the points using given string
    for (size_t itr = 0; itr < len_string_max; itr++)
    {
        if (data_string[itr] == '+') points += 1;
        if (data_string[itr] == '-') points -= 1;
    }

    // STEP.END
    return points;
}

int main (void)
{
    // variables for `main`
    char given_string[len_string_max];
    int  rtvl_scanf;

    // STEP.01
    // read out the given string
    rtvl_scanf = scanf("%s", given_string);

    // STEP.02
    // calcualte & output the points
    printf("%d\n", calculate_points(given_string));

    // STEP.END
    return 0;
}

File: Haskell/ABC/ABC101/ABC101_A_v01.hs
-- [contest]  ABC 101
-- [task]     A
-- [URL]      https://atcoder.jp/contests/abc101/tasks/abc101_a
-- [compiler] Haskell (GHC 7.10.3)
-- [status]   https://atcoder.jp/contests/abc101/submissions/6099349 : WA
--            https://atcoder.jp/contests/abc101/submissions/6099364 : AC

-- [reference]
-- http://hackage.haskell.org/package/bytestring-0.10.8.2/docs/Data-ByteString-Char8.html
-- https://qiita.com/hnw/items/3f7d27b742c5a1a99a9a
-- https://myuon.github.io/posts/haskell-atcoder/

-- modules to import --

import Data.ByteString.Char8
import Prelude

-- functions for this task --

convert_sign :: ByteString -> Int
convert_sign string_given
        | Data.ByteString.Char8.head string_given == '+' =  1
        | Data.ByteString.Char8.head string_given == '-' = -1

task_A :: ByteString -> Int
task_A string_given
        | Data.ByteString.Char8.length string_given > 0 = (convert_sign $ string_given) + (task_A $ Data.ByteString.Char8.tail $ string_given)
        | otherwise                                     = 0

-- main process is below --

main :: IO ()
main = do

        -- STEP.01
        -- read out the given string
        string_given <- Data.ByteString.Char8.getLine

        -- STEP.02
        -- output the answer of this task
        Prelude.print $ task_A $ string_given

-- End of this source code --
